### DISCUSSION TITLE
Add graph-level features for graph regression?

### DISCUSSION THREAD

**Discussion by Shualdon** [pyg-team/pytorch_geometric#4433] [Q&A]
Hi,

I am working on a graph regression prediction model and in addition to the node and edge features I want to add graph features.
I tried to search and couldn't find any information if that's possible, or if there is anyone who did something similar.
I was thinking of adding 'graph_attr' feature vector to the Data object and then concat it to the Pooling layer of the graph.
Does that sound like it's a good idea? Or is there some other way to do so?

Thanks.

**Comment by rusty1s** [comment]
Yes, this is a good idea. You can also try to make this information already available during message passing, e.g., via
```python
data.x = torch.cat([data.x, data.graph_attr[data.batch]], dim=-1)
```
